milestone_1/crop_yield/crop_yield.md

```diff
@@ -0,0 +1,297 @@
+from pyproj import Proj
+```
+
+```python
+url = "http://download.geofabrik.de/europe/netherlands/gelderland-latest-free.shp.zip"
+extraction = "shape_files"
+if not os.path.exists(extraction):
+    os.makedirs(extraction)
+
+response = requests.get(url)
+if response.status_code == 200:
+    zipfile = zipfile.ZipFile(io.BytesIO(response.content))
+    zipfile.extractall(extraction)
+    print("Shape files downloaded")
+    print("Shape files are in shape_files")
+```
+
+```python
+gdf = gpd.read_file("shape_files/gis_osm_natural_free_1.shp")
+farthest_west = gdf.geometry.bounds["minx"].idxmin()
+
+# Find the farthest east point
+farthest_east = gdf.geometry.bounds["maxx"].idxmax()
+farthest_north = gdf.geometry.bounds["maxy"].idxmax()
+farthest_south = gdf.geometry.bounds["miny"].idxmin()
+west_coordinates = gdf.loc[farthest_west, "geometry"].coords[0]
+east_coordinates = gdf.loc[farthest_east, "geometry"].coords[0]
+north_coordinates = gdf.loc[farthest_north, "geometry"].coords[0]
+south_coordinates = gdf.loc[farthest_south, "geometry"].coords[0]
+print("Farthest West Coordinates:", west_coordinates)
+print("Farthest East Coordinates:", east_coordinates)
+print("Farthest North Coordinates:", north_coordinates)
+print("Farthest South Coordinates:", south_coordinates)
+```
+
+```python
+##Read Files using geopandas
+print(gdf.head())
+```
+
+```python
+"""""" """
+Brief Intro: MODIS assumes the Earth is a sphere
+with a R = 6.3781e6 m (6371007.181 m) radius.
+The world is subdivided into 18 vertical tiles and 36 horizontal tiles.
+each tile has 2400 cells of 500m x 500m.
+"""
+# Constants
+CELLS = 2400
+VERTICAL_TILES = 18
+HORIZONTAL_TILES = 36
+EARTH_RADIUS = 6371007.181
+EARTH_WIDTH = 2 * math.pi * EARTH_RADIUS
+
+# Calculate tile and cell sizes
+TILE_WIDTH = EARTH_WIDTH / HORIZONTAL_TILES
+TILE_HEIGHT = TILE_WIDTH
+CELL_SIZE = TILE_WIDTH / CELLS
+
+
+# Define a projection for the MODIS sinusoidal grid
+MODIS_GRID = Proj(f"+proj=sinu +R={EARTH_RADIUS} +nadgrids=@null +wktext")
+
+# Function to convert latitude and longitude to MODIS grid coordinates
+
+
+def lat_lon_to_modis(lat, lon):
+    # Use the MODIS_GRID projection to convert
+    # latitude and longitude to x, y coordinates
+    x, y = MODIS_GRID(lon, lat)
+
+    # Calculate horizontal and vertical grid indices (h and v)
+    h = (EARTH_WIDTH * 0.5 + x) / TILE_WIDTH
+    v = -(EARTH_WIDTH * 0.25 + y - (VERTICAL_TILES - 0) * TILE_HEIGHT) / TILE_HEIGHT
+
+    # Return the grid indices as integers
+    return int(h), int(v)
+
+
+# Example usage
+x, y = lat_lon_to_modis(51.7422767, 5.9325433)
+# 5.0115545, 51.8170254
+# 6.8113423, 51.9661496
+# 5.9325433, 51.7422767
+print(x)
+print(y)
+```
+
+```python
+"""""" """
+Product must be same name from the list at https://e4ftl01.cr.usgs.gov/MOLT/
+""" ""
+# Define the parameters
+product = "MOD09A1.061"  # e.g., 'MOD09A1'
+tiles = "h18v03"  # e.g., 'h17v03,h18v03'
+# Gelderland falls in 18th Horizontally and third vertically.
```

Comment:
It would be terrific to understand how you got to this conclusion. You can just add it in markdown in the notebook.